### PR TITLE
chore: Add Ninja generator for wasm & document Ninja requirement for presets

### PR DESCRIPTION
**Pull request by phated**
# Description

At the end of #162, @codygunton said that the wasm preset was slow in VSCode. That is likely due to not using the Ninja generator that @adr1anh added for the default preset (the wasm preset doesn't inherit from default since it is drastically different).

This sets the wasm preset to use the Ninja generator to match the default generator and it documents Ninja as a dependency for the preset workflow. If a user doesn't use the presets for some reason, they will continue to use whatever generator they specify (or their default).

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

**Comment by phated** [comment]
Looks like Ninja doesn't exist in the wasm docker container. I'll fix.

File: cpp/dockerfiles/Dockerfile.wasm-linux-clang [hunk 1]
```diff
@@ -1,5 +1,5 @@
 FROM ubuntu:kinetic AS builder
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y build-essential wget git libssl-dev cmake curl binaryen
```
Comment (phated) [inline cpp/dockerfiles/Dockerfile.wasm-linux-clang:2]: Need to figure out why ninja isn't available here 🤔

**Review by codygunton — APPROVED** [review]
Tested this out, works for me, and is much faster than building without ninja via CMake Tools in VS Code.

File: cpp/dockerfiles/Dockerfile.wasm-linux-clang (hunk 1 above)
Comment (phated) [reply cpp/dockerfiles/Dockerfile.wasm-linux-clang:2]: Ubuntu names things weirdly: https://packages.ubuntu.com/kinetic/ninja-build

**Pull request merged**